app/Console/Commands/FlightawareImport.php

```diff
@@ -98,16 +98,17 @@ public function load_departures(&$flights, $ratelimit_max_flights, $ratelimit_ti
         // First, load as many pages of departures as is configured
         for ($i = 0; $i < intval(Config::get('flightaware.departure_pages_per_chunk')); $i++) {
             // If we hit the ratelimit, pause for a bit.
-            if ($flights_count > $ratelimit_max_flights) {
-                $pbar->setMessage('Ratelimit timeout');
-                $pbar->advance();
-                $flights_count = 0;
-                sleep($ratelimit_timeout);
-            }
+            // Commenting out the ratelimit timeout logic - this needs to be rebuilt to consider a chunk returning zero flights
+            //if ($flights_count > $ratelimit_max_flights) {
+            $pbar->setMessage('Ratelimit timeout');
+            $pbar->advance();
+            $flights_count = 0;
+            sleep($ratelimit_timeout);
+            //}
```

Comment:
Is this something to be reworked before merging or for the future? If this is for the future, I'd prefer for this to be a screenshot in a new story and have this comment removed

Reply:
@iccowan I'm concerned about how much code we'd be orphaning without any explanation if we did this. The rate limiter is a key piece that keeps the price of these API calls reasonable. It is absolutely critical that we fix it before the next event. If I screenshot and remove this, the calculations behind $flights_count and $ratelimit_max_flights will no longer make sense.